packages/spacecat-shared-rum-api-client/src/functions/form-vitals.js

```diff
@@ -133,8 +134,62 @@ function containsFormVitals(row) {
   return METRICS.some((metric) => Object.keys(row[metric]).length > 0);
 }
 
+function getParentPageVitalsGroupedByIFrame(bundles, dataChunks, iframeParentMap) {
+  const iframeVitals = {};
+  if (dataChunks.facets.urlUserAgents) {
+    dataChunks.facets.urlUserAgents.reduce((acc, { value, weight }) => {
```

Comment:
pls move this logic inside the existing code where we are generating form vitals by iterating over  `dataChunks.facets.urlUserAgents`